transport/src/main/java/io/netty/channel/FileRegion.java

```diff
@@ -84,9 +86,86 @@
      *                  transfer start from {@link #position()}th byte and
      *                  <tt>{@link #count()} - 1</tt> will make the last
      *                  byte of the region transferred.
+     * @deprecated      use {@link #transferIndex()} based API.
      */
+    @Deprecated
     long transferTo(WritableByteChannel target, long position) throws IOException;
 
+    /**
+     * Transfers the content of this file region to the specified channel.<p>
+     * This method does not modify {@link #transferIndex()}.
+     *
+     * @param target    the destination of the transfer
+     * @param position  the relative offset of the file where the transfer
+     *                  begins from.  For example, <tt>0</tt> will make the
+     *                  transfer start from {@link #position()}th byte and
+     *                  <tt>{@link #count()} - 1</tt> will make the last
+     *                  byte of the region transferred.
+     * @param length    the maximum number of bytes to transfer
+     */
+    long transferBytesTo(WritableByteChannel target, long position, long length) throws IOException;
```

Comment:
We can not add methods to the interface until we release a new minor / major version. So please remove this and just check for the actual implementation with an instanceof when you need these methods.